app/entry.server.tsx

```diff
@@ -29,24 +35,33 @@ export default async function handleRequest(
 		? 'onAllReady'
 		: 'onShellReady'
 
+  const nonce = String(loadContext.cspNonce) ?? undefined
 	return new Promise((resolve, reject) => {
 		let didError = false
 
 		const { pipe, abort } = renderToPipeableStream(
-			<RemixServer context={remixContext} url={request.url} />,
+      <NonceProvider value={nonce}>
+        <RemixServer context={remixContext} url={request.url} />
+      </NonceProvider>,
 			{
-				[callbackName]: () => {
-					const body = new PassThrough()
-
+        nonce,
+        [callbackName]: () => {
+          const body = new PassThrough()
 					responseHeaders.set('Content-Type', 'text/html')
+          const dataEvtTransform = new Transform({
+            transform(chunk, _, callback) {
+              const string = chunk.toString()
+              const replaced = string.replace(/data-evt-/g, `nonce="${nonce}" `)
+              callback(null, replaced)
+            },
+          })
+          pipe(dataEvtTransform).pipe(body)
```

Comment:
Let's not bother with this. This is something I'm doing in my own site that's unnecessary for anyone else.